Mixpanel/Mixpanel.m

```diff
@@ -651,14 +651,20 @@ - (void)flushWithCompletion:(void (^)())handler
 
 - (void)flushEvents
 {
-    [self flushQueue:_eventsQueue
-            endpoint:@"/track/"];
+    NSMutableArray *events = [self unarchiveEvents];
+    [self flushQueue:events
+            endpoint:kEndpointTrack];
+    
+    [self archiveAllEvents:events];
```

Comment:
Following my first comment, this is a good candidate to change as well. `flushEvents` will just flush and can return the events that are not flushed. Those events will be captured on `flushWithCompletion` that will just call `archive`.